### ISSUE TITLE
Switch to openbao docker images

### ISSUE THREAD

**Issue by jessebot** [openbao/openbao-helm#3]
## Description
I would like to replace all the references to hashicorp/vault docker images with our new official images.

## Context
This is dependent on the new docker images being uploaded, which is blocked by dockerhub recognizing openbao as an opensource project to raise limit caps. When that is completed, we can start on a PR to get this done. Ref: https://github.com/openbao/openbao/issues/327#issuecomment-2099246264

There's a number of references to official hashicorp vault images and their versions, taken from the newly generated docs for this repo in #2 :

| parameter                      | type   | default value                    |
|--------------------------------|--------|----------------------------------|
| csi.agent.image.repository     | string | `"hashicorp/vault"`              |
| csi.agent.image.tag            | string | `"1.15.2"`                       |
| csi.image.repository           | string | `"hashicorp/vault-csi-provider"` |
| csi.image.tag                  | string | `"1.4.1"`                        |
| injector.agentImage.repository | string | `"hashicorp/vault"`              |
| injector.agentImage.tag        | string | `"1.15.2"`                       |
| injector.image.repository      | string | `"hashicorp/vault-k8s"`          |
| injector.image.tag             | string | `"1.3.1"`                        |
| server.image.repository        | string | `"hashicorp/vault"`              |
| server.image.tag               | string | `"1.15.2"`                       |

It looks like we need a replacement for:

- [ ] `hashicorp/vault` <-- we will have this soon after dockerhub is up
- [ ] `hashicorp/vault-csi-provider` <-- needs it's official repo forked
- [ ] `hashicorp/vault-k8s` <-- not sure if this is available, have to hunt it down

**Comment by naphelps** [comment]
@jessebot Quay.io is also available today: https://quay.io/repository/openbao/openbao

We will also be using GitHub builtin container registry as well in the future.

**Comment by jessebot** [comment]
Thank you, @naphelps for the prompt responses :)

I can submit a PR for the `hashicorp/vault` -> `quay.io/openbao/openbao` in just a bit.

We'll still need to wait on the other two images though.

**Comment by jessebot** [comment]
I will start work on this today to get the base image references updated.

**Comment by jessebot** [comment]
Updates: 

- With the merge of #8, we got 60-66.66% of this Issue completed. Since  https://github.com/openbao/openbao-csi-provider was recently forked into the org, I can do another PR updating those images. 

- Still unsure of the status of the `hashicorp/vault-k8s` though.

**Comment by naphelps** [comment]
https://github.com/openbao/openbao-k8s

**Comment by jessebot** [comment]
Thanks, @naphelps! Looks like both of those repos could use some work I'll try to take a look in the next day or so to see what they need to get their search/replace done and docker images released via ghcr, docker hub, or quay. If others in the community beat me to it, please feel free to comment here and let me know which registry to point to :)

**Comment by MagicRB** [comment]
For anyone wondering why this helm chart seems to do nothing, it's because it's still using `hashicorp/vault-k8s` which has different annotation keys...

**Comment by cipherboy** [comment]
@MagicRB Correct, we've not yet created a release version of this helm chart yet, it is still under development. :-) One of the things holding this up is a new OpenBao release, as it turns out we didn't have the correct Kubernetes auto-discovery tags in the 2.0.0 GA release. 

Please bear with us, or if you have the expertise, PRs are welcome :-) But I think mostly a new OpenBao Core release is held up on transactional storage review & merge.

**Comment by MagicRB** [comment]
No worries :) im bearing with you, just wanted to leave it as a warning for early explorers haha. I managed to get it working, the injector is still hashicorp, but the agent id a custom built container which contains openbao (with a symlink from vault, to make it work). I will opena a few PRs making the chart behave a bit nicer on my infra, since I need to pass in images of the form `nix:0/nix/store/aaaaaa-image` and currently the chart always expects a tag to bethere so I had to hack around it slightly.

**Comment by shaeliss** [comment]
What is the current way to use the injector agent? Is it working with openbao agentImage and openbao annotations in the yaml file? Or the only way for now is to use vault agentImage and vault annotations?

**Comment by eyenx** [comment]
`openbao-csi-provider` has been addressed in #65
